### PR TITLE
63 additional shutdown signals

### PR DESCRIPTION
**Pull request by wolf4ood**
Hi,

I've created 2 versions of shutdown:

- Unix SIGTERM/ SIGINT (with logging)
- Not Unix only ctrl-c (without logging)

### CODE REVIEW

File: src/bin/toshi.rs [hunk 1]
```diff
@@ -205,15 +205,36 @@ fn connect_to_consul(settings: &Settings) -> impl Future<Item = (), Error = ()>
...
+    S: Stream<Item = String, Error = std::io::Error>,
+    F: Fn(&str),
+{
+    stream
         .take(1)
         .into_future()
-        .and_then(move |_| {
+        .and_then(move |(sig, _)| {
+            if let Some(s) = sig {
+                logger(&s)
```
Comment (LucioFranco) [inline src/bin/toshi.rs]: Could you explain why you decided to use a `Fn(&str)` over just calling `info!`?

Comment (wolf4ood) [reply src/bin/toshi.rs]: @LucioFranco  done :)

**Comment by LucioFranco** [comment]
@wolf4ood one thing left it looks like you need to run a rustfmt pass through it as CI check it and its showing a diff https://travis-ci.org/toshi-search/Toshi/jobs/481621163#L758

**Comment by LucioFranco** [comment]
Great thank you! 👍

**Pull request merged**